nRFMeshProvision/MeshNetworkManager.swift

```diff
@@ -332,19 +323,15 @@ public extension MeshNetworkManager {
     ///   - initialTtl:     The initial TTL (Time To Live) value of the message.
     ///                     If `nil`, the default Node TTL will be used.
     ///   - applicationKey: The Application Key to sign the message.
-    ///   - completion:     The completion handler called when the message
-    ///                     has been sent.
     /// - throws: This method throws when the mesh network has not been created,
     ///           the local Node does not have configuration capabilities
     ///           (no Unicast Address assigned), or the given local Element
-    ///           does not belong to the local Node.
-    /// - returns: Message handle that can be used to cancel sending.
-    @discardableResult
+    ///           does not belong to the local Node, or the manager failed to
+    ///           send the message.
     func send(_ message: MeshMessage,
               from localElement: Element? = nil, to destination: MeshAddress,
               withTtl initialTtl: UInt8? = nil,
-              using applicationKey: ApplicationKey,
-              completion: ((Result<Void, Error>) -> ())? = nil) throws -> MessageHandle {
+              using applicationKey: ApplicationKey) async throws {
```

Comment:
may need api of form 
`func send(message: MeshMessage,  localElementIdx: UInt8?=nil, destination: MeshAddress, initialTtl:, applicationKeyIdx: KeyIndex) async throws {`   
such that synchronous code that makes the call from within a new Task where the task captures required Sendable method parameters.

Reply:
Same as above.